File: 2023/UICPC DIV1/UICPC Round #05/solutions/froshweek2/submissions/accepted/froshweek2-zf.py
#!/usr/bin/python3

n, m = map(int, input().split())
tasks = list(map(int, input().split()))
quiet = list(map(int, input().split()))

tasks.sort()
quiet.sort()

n -= 1
m -= 1
tot = 0
while n >= 0 and m >= 0:
    if tasks[n] <= quiet[m]:
        m -= 1
        tot += 1
    n -= 1
print(tot)

File: 2023/UICPC DIV1/UICPC Round #05/solutions/froshweek2/submissions/accepted/froshweek2-zf.cpp
#include <iostream>
#include <algorithm>

#define MAXN 200000

using namespace std;

int main() {
    int n, m, tot = 0;
    int task[MAXN], quiet[MAXN];

    cin >> n >> m;
    for (int i = 0; i < n; ++i) cin >> task[i];
    for (int i = 0; i < m; ++i) cin >> quiet[i];

    sort(task, task + n);
    sort(quiet, quiet + m);

    --n;
    --m;

    while (n >= 0 && m >= 0) {
        if (task[n] <= quiet[m]) {
            ++tot;
            --m;
        }
        --n;
    }

    cout << tot << endl;

    return 0;
}
